src/backend/distributed/commands/function.c

```diff
@@ -1494,8 +1516,10 @@ PreprocessAlterFunctionStmt(Node *node, const char *queryString,
 	AlterFunctionStmt *stmt = castNode(AlterFunctionStmt, node);
 	AssertObjectTypeIsFunctional(stmt->objtype);
 
-	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt, false);
-	if (!ShouldPropagateAlterFunction(&address))
+	List *addresses = GetObjectAddressFromParseTree((Node *) stmt, false);
+	Assert(list_length(addresses) == 1);
+
+	if (!ShouldPropagateAlterFunction(addresses))
```

Comment:
similar to `CreateAllFunctionDDLCommandsIdempotent `, I'd rather drop very specific functions such as `ShouldPropagateAlterFunction` which ever only accepts a single address.

Postgres syntax does not allow altering multiple functions in a single statement anyway.